### ISSUE TITLE
Setting arguments for transformers.TrainingArguments that do not exist.

### ISSUE THREAD

**Issue by taoshende** [Polygl0t/llm-foundry#72]
### System Info

Bender, A40. Installed the environment via the installation script `alignment/slurm/create_venv_bender.sh`:
```
torch==2.6.0+cu124 (runtime cuda=12.4)
trl==1.13.0
transformers==5.17.0
datasets==5.0.1
```

### Information

- [x] An existing script in the repository (e.g. under `distributed/`, `data/`, `evals/`)
- [ ] My own modified or new scripts

### Area

- [ ] Pretraining (`distributed/`)
- [x] Post-training / alignment (`alignment/`)
- [ ] Data processing / tokenization (`data/`)
- [ ] Evaluation (`evals/`)
- [ ] Synthetic data (`synthetic/`)
- [ ] Model merging (`merge/`)
- [ ] Tokenizer (`tokenizer/`)
- [ ] Other / tools (`tools/`)

### Reproduction

In `alignment/dpo_trainer.py` we set up the training arguments in `trl.DPOConfig`.
Some of the arguments we pass do not exist [link to docs](https://huggingface.co/docs/transformers/main/en/main_classes/trainer#transformers.TrainingArguments):
- `max_prompt_length` and `max_completion_length`. I suspect that they are redundant as we already pass `max_length`. It could be that they were taken on from `trl.GRPOConfig` in `grpo_trainer.py` where they are needed.
- `label_pad_token_id`
- `include_tokens_per_second`. This also concerns `sft_*`, `grpo_*`, and `reward_trainer.py`.

### Suggested Fixes

I suggest to remove the parameters.
However, I am not sure what the indented behavior behind the parameters was, and whether we want alternatives.

I have tested `dpo_trainer.py` with the parameters above removed.

**Comment by mira687** [comment]
Both conclusions are right, but for two different reasons — which matters for your "I am not sure what the intended behavior was" question.

`max_prompt_length`, `max_completion_length` and `label_pad_token_id` were never `transformers.TrainingArguments` fields; they were `DPOConfig` fields that TRL dropped in the DPO refactor. At `v1.13.0`, [`dpo_config.py`](https://github.com/huggingface/trl/blob/v1.13.0/trl/trainer/dpo_config.py) keeps only `max_length` (default 1024) and `truncation_mode`, and none of the three names appear anywhere in `dpo_config.py` or `dpo_trainer.py`. `include_tokens_per_second` is the opposite case: that one really was a `TrainingArguments` field, and transformers removed it — zero hits in `training_args.py` at `v5.17.0`.

So removing them is the fix, but `max_prompt_length` isn't a no-op. Truncation now happens in `DataCollatorForPreference` on the concatenated prompt+completion, sliced to `max_length`. With the defaults in `dpo_trainer.py` (`max_length=4096`, `max_prompt_length=1024`) the prompt goes from a hard 1024-token cap to whatever is left of 4096, so long-prompt pairs will train on considerably more context than they did before.

Over-long prompts are also dropped now rather than truncated:

```python
# trl/trainer/dpo_trainer.py, _prepare_dataset
if args.max_length is not None and args.truncation_mode == "keep_start":
    dataset = dataset.filter(lambda example: len(example["prompt_ids"]) < args.max_length, **map_kwargs)
```

Your `--truncation_mode` default is `keep_end`, so nothing is dropped today. But the `DPOConfig` docstring marks `keep_end` deprecated and slated for removal in v2.0.0, and when that lands any pair whose prompt alone reaches `max_length` leaves the training set with no error and only a progress-bar label to show for it. Logging `len(train_dataset)` either side of trainer construction would make that visible if it ever starts happening.
